src/nlp/totext.ts

```diff
@@ -104,115 +120,150 @@ export default class ToText {
           days.indexOf('TH') !== -1 &&
           days.indexOf('FR') !== -1 &&
           days.indexOf('SA') !== -1 &&
-          days.indexOf('SU') !== -1
+          days.indexOf('SU') !== -1,
       }
 
       const sortWeekDays = function (a: Weekday, b: Weekday) {
         return a.weekday - b.weekday
       }
 
-      this.byweekday.allWeeks!.sort(sortWeekDays)
-      this.byweekday.someWeeks!.sort(sortWeekDays)
+      this.byweekday.allWeeks.sort(sortWeekDays)
+      this.byweekday.someWeeks.sort(sortWeekDays)
 
-      if (!this.byweekday.allWeeks!.length) this.byweekday.allWeeks = null
-      if (!this.byweekday.someWeeks!.length) this.byweekday.someWeeks = null
+      if (!this.byweekday.allWeeks.length) this.byweekday.allWeeks = null
+      if (!this.byweekday.someWeeks.length) this.byweekday.someWeeks = null
     } else {
       this.byweekday = null
     }
   }
 
   /**
    * Test whether the rrule can be fully converted to text.
+   *
    * @param {RRule} rrule
    * @return {Boolean}
    */
-  static isFullyConvertible (rrule: RRule) {
-    let canConvert = true
+  static isFullyConvertible(rrule: RRule) {
+    const canConvert = true
 
     if (!(rrule.options.freq in ToText.IMPLEMENTED)) return false
     if (rrule.origOptions.until && rrule.origOptions.count) return false
 
-    for (let key in rrule.origOptions) {
+    for (const key in rrule.origOptions) {
       if (contains(['dtstart', 'wkst', 'freq'], key)) return true
       if (!contains(ToText.IMPLEMENTED[rrule.options.freq], key)) return false
     }
 
     return canConvert
   }
 
-  isFullyConvertible () {
+  isFullyConvertible() {
     return ToText.isFullyConvertible(this.rrule)
   }
 
   /**
    * Perform the conversion. Only some of the frequencies are supported.
    * If some of the rrule's options aren't supported, they'll
    * be omitted from the output an "(~ approximate)" will be appended.
+   *
+   * @param toTextOptions
    * @return {*}
    */
-  toString () {
+  toString(toTextOptions: { condenseOutput?: boolean } = {}) {
     const gettext = this.gettext
 
-    if (!(this.options.freq! in ToText.IMPLEMENTED)) {
+    if (!(this.options.freq in ToText.IMPLEMENTED)) {
       return gettext('RRule error: Unable to fully convert this rrule to text')
     }
 
-    this.text = [gettext('Repeats every')]
+    const { condenseOutput } = toTextOptions
```

Comment:
Relevant changes in this `toString` function and relevant `DAILY` `MONTHLY` `YEARLY` etc functions